### PR TITLE
Aggregate rules documentation

### LINKED ISSUE #342: Documentation for aggregate rules
We need to document the new aggregate rules for contributors of built-in rules as well as for authors of custom rules. Things to include:

1. How to author rules using aggregated data.
2. When and when not Regal considers aggregate rules (see #340)
3. Helper functions or helper rules available.
4. A simple example.

### PR DESCRIPTION
**Pull request by anderseknert**
I'm sure there's more to it, but we can iterate on this when we hear back from the first custom aggregate rules authors.

Fixes #342

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,6 +190,16 @@ The following rules are currently available:
 
 By default, all rules except for those in the `custom` category are currently **enabled**.
 
+#### Aggregate Rules
+
+Most Regal rules will use data only from a single file at a time, with no consideration for other files. A few rules
+however require data from multiple files, and will therefore collect, or aggregate, data from all files provided for
+linting. These rules are called *aggregate rules*, and will only be run when there are more than one file to lint, such
```
Comment (charlieegan3) [inline README.md:197]:
```suggestion
linting. These rules are called *aggregate rules*, and will only be run when there is more than one file to lint, such
```

File: docs/custom-rules.md [hunk 2]
```diff
@@ -164,6 +164,83 @@ Starting from top to bottom, these are the components comprising our custom rule
    will later be included in the final report provided by Regal.
 1. The `result.location` helps extract the location from the element failing the test. Make sure to use it!
 
+## Aggregate Rules
+
+Aggregate rules are a special type of rule that allows you to collect data from multiple files before making a decision.
+This is needed in some cases where a single policy won't be enough for a linter rule to make a decision. For example,
```
Comment (charlieegan3) [inline docs/custom-rules.md:170]:
```suggestion
This is needed in some cases where a single policy file won't be enough for a linter rule to make a decision. For example,
```

File: docs/custom-rules.md [hunk 3]
```diff
@@ -164,6 +164,83 @@ Starting from top to bottom, these are the components comprising our custom rule
...
+you may want to enforce that in a policy repository, there must be at least one package annotated with an `authors`
+attribute. This requires first collecting annotation data from all the provided policies, and then have the rule use
+this data to make a decision. The structure of an aggregate rule is similar to a regular rule, but with a few notable
+differences. The most significant one is that linting happens in two separate phases — one that aggregates data from
+files, and one that actually lints and reports violations using that data.
```
Comment (charlieegan3) [inline docs/custom-rules.md:175]: Take or leave this one, I thought it made sense to strictly tie the explanation to the two rules in the code below.

```suggestion
files (`aggregate`), and one that actually lints and reports violations using that data (`aggregate_report`).
```

**Review by charlieegan3 — APPROVED** [review]

**Comment by anderseknert** [comment]
Thanks for the review @charlieegan3 👍 I applied all of your suggestions before merging.

**Pull request merged**